warehouse/query-core/src/test/java/datawave/query/jexl/visitors/TreeFlatteningRebuildingVisitorTest.java

```diff
@@ -166,19 +169,42 @@ public void multipleNestingMixedOpsTest() throws Exception {
     }
     
     @Test
-    public void singleChildAndOrTest() throws Exception {
-        JexlNode eqNode = JexlASTHelper.parseJexlQuery("FIELD == 'value'");
+    public void singleChildAndOrTest() {
+        JexlNode eqNode = JexlNodeFactory.buildEQNode("FIELD", "value");
         JexlNode and1 = JexlNodeFactory.createUnwrappedAndNode(Collections.singleton(eqNode));
         JexlNode or1 = JexlNodeFactory.createUnwrappedOrNode(Collections.singleton(and1));
         
         JexlNode flattened = TreeFlatteningRebuildingVisitor.flatten(or1);
         
-        Assert.assertEquals(ASTJexlScript.class, flattened.getClass());
-        Assert.assertEquals(1, flattened.jjtGetNumChildren());
-        Assert.assertEquals(ASTEQNode.class, flattened.jjtGetChild(0).getClass());
+        Assert.assertEquals(ASTEQNode.class, flattened.getClass());
+        Assert.assertEquals(2, flattened.jjtGetNumChildren());
         Assert.assertEquals(JexlStringBuildingVisitor.buildQuery(eqNode), JexlStringBuildingVisitor.buildQuery(flattened));
     }
     
+    @Test
+    public void testWrappedUnionWithSingleChild() throws ParseException {
+        // have to build this up manually. Creating an Or node via JexlNodeFactory#createOrNode is smart enough
+        // to not wrap single children
+        JexlNode eq = JexlNodeFactory.buildEQNode("FIELD", "value");
+        
+        JexlNode union = new ASTOrNode(ParserTreeConstants.JJTORNODE);
+        JexlNodes.children(union, eq);
+        
+        JexlNode refExpr = JexlNodes.wrap(union);
+        JexlNode ref = JexlNodes.makeRef(refExpr);
+        ASTJexlScript script = JexlNodeFactory.createScript(ref);
+        
+        JexlNode flattened = TreeFlatteningRebuildingVisitor.flatten(script);
+        
+        Assert.assertEquals("FIELD == 'value'", JexlStringBuildingVisitor.buildQueryWithoutParse(flattened));
```

Comment:
If I remove your changes to TreeFlatteningRebuilder, then the actual query string in this test becomes `(FIELD == 'value')`.  That is the expected behavior though when you call `flatten`.  If you call `flattenAll`, the reference expression is removed and the test passes.